lib/cleanup_helper.rb

```diff
@@ -50,14 +50,18 @@ def clean_puppet
         ActiveRecord::Base.connection.remove_column :hostgroups, :environment_id
       end
       ActiveRecord::Base.connection.drop_table(:environments, if_exists: true, force: :cascade)
+      ActiveRecord::Base.connection.drop_table(:hostgroup_puppet_facets, if_exists: true, force: :cascade)
+      ActiveRecord::Base.connection.drop_table(:host_puppet_facets, if_exists: true, force: :cascade)
 
       ActiveRecord::SchemaMigration.
         where(version: %w[20090722141107 20090802062223 20110412103238 20110712070522
                           20120824142048 20120905095532 20121018152459 20130725081334
                           20140318153157 20140407161817 20140407162007 20140407162059
                           20140413123650 20140415032811 20141109131448 20150614171717
                           20160307120453 20160626085636 20180831115634 20181023112532
-                          20181224174419]).delete_all
+                          20181224174419 20090905150132 20101121140000 20201125113903
+                          20200803113803 20200722171017 20200803113531 20200803113903
+                          20211111125003]).delete_all
```

Comment:
Similar, I tried to find these files. They aren't in Foreman:
```console
$ ls db/migrate/20181224174419_* db/migrate/20090905150132_* db/migrate/20101121140000_* db/migrate/20201125113903_* db/migrate/20200803113803_* db/migrate/20200722171017_* db/migrate/20200803113531_* db/migrate/20200803113903_* db/migrate/20211111125003_* 
ls: cannot access 'db/migrate/20181224174419_*': No such file or directory
ls: cannot access 'db/migrate/20090905150132_*': No such file or directory
ls: cannot access 'db/migrate/20101121140000_*': No such file or directory
ls: cannot access 'db/migrate/20201125113903_*': No such file or directory
ls: cannot access 'db/migrate/20200803113803_*': No such file or directory
ls: cannot access 'db/migrate/20200722171017_*': No such file or directory
ls: cannot access 'db/migrate/20200803113531_*': No such file or directory
ls: cannot access 'db/migrate/20200803113903_*': No such file or directory
ls: cannot access 'db/migrate/20211111125003_*': No such file or directory
```

But some are in puppet_foreman:
```console
$ ls -1 db/migrate/20181224174419_* db/migrate/20090905150132_* db/migrate/20101121140000_* db/migrate/20201125113903_* db/migrate/20200803113803_* db/migrate/20200722171017_* db/migrate/20200803113531_* db/migrate/20200803113903_* db/migrate/20211111125003_* 
ls: cannot access 'db/migrate/20181224174419_*': No such file or directory
ls: cannot access 'db/migrate/20090905150132_*': No such file or directory
db/migrate/20101121140000_add_environment_to_template_combinations.foreman_puppet.rb
db/migrate/20200722171017_create_host_puppet_facet.foreman_puppet.rb
db/migrate/20200803113531_create_hostgroup_puppet_facet.foreman_puppet.rb
db/migrate/20200803113803_migrate_environment_to_puppet_facet.foreman_puppet.rb
db/migrate/20200803113903_migrate_host_type_in_host_config_groups.foreman_puppet.rb
db/migrate/20201125113903_migrate_puppetclasses_to_facets.foreman_puppet.rb
db/migrate/20211111125003_drop_puppetclasses_direct_references.foreman_puppet.rb
```

That begs the question: where are `20090905150132` and `20181224174419` from?

Reply:
Hello, 
https://github.com/theforeman/foreman_puppet/tree/master/db/migrate_foreman and https://github.com/theforeman/foreman_puppet/tree/master/db/migrate has the related migration files

Reply:
Ah thanks. A comment in this file about how to find these magic numbers will be nice for any future reader.

Reply:
Thanks. I will keep this point documented as a best practice..